spec/commands/deploy.spec.ts

```diff
@@ -0,0 +1,393 @@
+      │         ggt v0.3.3                                                   │
+      │                                                                      │
+      │         App         test                                             │
+      │         Editor      https://test.gadget.app/edit                     │
+      │         Playground  https://test.gadget.app/api/graphql/playground   │
+      │         Docs        https://docs.gadget.dev/api/test                 │
+      │                                                                      │
+      │         Endpoints                                                    │
+      │           • https://test.gadget.app                                  │
+      │           • https://test--development.gadget.app                     │
+      │                                                                      │
+      ╰──────────────────────────────────────────────────────────────────────╯
+
+      Building frontend assets ...
+
+      Setting up database ...
+
+      Copying development ...
+
+      Restarting app ...
+
+      Deploy completed. Good bye!
+      "
+    `);
+  });
+
+  it("can not deploy if the maximum number of applications has been reached", async () => {
+    await makeSyncScenario({ localFiles: { ".gadget/": "" } });
+
+    const mockEditGraphQL = makeMockEditGraphQLSubscriptions();
+    const error = new EditGraphQLError(REMOTE_SERVER_CONTRACT_STATUS_SUBSCRIPTION, [
+      {
+        message: "GGT_PAYMENT_REQUIRED: Payment is required for this application.",
+      },
+    ]);
+
+    await deploy(ctx);
+    const publishStatus = mockEditGraphQL.expectSubscription(REMOTE_SERVER_CONTRACT_STATUS_SUBSCRIPTION);
+
+    publishStatus.emitError(error);
+
+    expectStdout().toMatchInlineSnapshot(`
+      "╭──────────────────────────────────────────────────────────────────────╮
+      │                                                                      │
+      │         ggt v0.3.3                                                   │
+      │                                                                      │
+      │         App         test                                             │
+      │         Editor      https://test.gadget.app/edit                     │
+      │         Playground  https://test.gadget.app/api/graphql/playground   │
+      │         Docs        https://docs.gadget.dev/api/test                 │
+      │                                                                      │
+      │         Endpoints                                                    │
+      │           • https://test.gadget.app                                  │
+      │           • https://test--development.gadget.app                     │
+      │                                                                      │
+      ╰──────────────────────────────────────────────────────────────────────╯
+      GGT_PAYMENT_REQUIRED: Payment is required for this application.
```

Comment:
we should make this message nicer

`Production environment limit reached. Upgrade you plan to deploy`